File: frontend/src/api/spin.ts
/**
 * SPIN API Client
 */
import axios from 'axios'

const API_BASE = '/api/v1'

export interface CompanyInfo {
  subject_type: 'company'
  background: string
  recent_news: string[]
  competitors: string[]
  potential_pains: string[]
  extra_info: Record<string, unknown>
}

export interface PersonInfo {
  subject_type: 'person'
  name: string
  title: string
  company: string
  background: string
  recent_activities: string[]
  potential_pains: string[]
  extra_info: Record<string, unknown>
}

export type InvestigationResult = CompanyInfo | PersonInfo

export interface CustomerContext {
  industry: string
  scale: string
  pain_points: string[]
}

export interface SpinQuestionsRequest {
  customer: CustomerContext
}

export interface SpinQuestionList {
  question_list_id: string
  situation_questions: string[]
  problem_questions: string[]
  implication_questions: string[]
  need_payoff_questions: string[]
  customer_context: CustomerContext
  created_at: string
}

export interface SpinQuestionResponse {
  success: boolean
  data?: SpinQuestionList
  error?: string
}

export const spinApi = {
  /**
   * 调查客户信息（公司或个人）
   */
  async investigate(customerName: string): Promise<InvestigationResult> {
    const response = await axios.post<{success: boolean; data?: InvestigationResult; error?: string}>(
      `${API_BASE}/spin/investigate`,
      { customer_name: customerName }
    )
    if (!response.data.success || !response.data.data) {
      throw new Error(response.data.error || '调查失败')
    }
    return response.data.data
  },

  /**
   * 生成SPIN问题清单
   */
  async generateQuestions(customer: CustomerContext): Promise<SpinQuestionList> {
    const response = await axios.post<SpinQuestionResponse>(`${API_BASE}/spin/questions`, {
      customer
    })
    if (!response.data.success || !response.data.data) {
      throw new Error(response.data.error || '生成失败')
    }
    return response.data.data
  },

  /**
   * 获取问题清单
   */
  async getQuestions(questionListId: string): Promise<SpinQuestionList> {
    const response = await axios.get<SpinQuestionResponse>(`${API_BASE}/spin/questions/${questionListId}`)
    if (!response.data.success || !response.data.data) {
      throw new Error(response.data.error || '获取失败')
    }
    return response.data.data
  },

  /**
   * 获取最新问题清单
   */
  async getLatestQuestions(): Promise<SpinQuestionList | null> {
    const response = await axios.get<SpinQuestionResponse>(`${API_BASE}/spin/questions`)
    if (response.data.success && response.data.data) {
      return response.data.data
    }
    return null
  }
}

export default spinApi

File: backend/app/schemas/spin.py
"""
SPIN Schemas - Pydantic models for SPIN question generation
"""
from pydantic import BaseModel, Field
from typing import List, Optional
from uuid import UUID


class CustomerContext(BaseModel):
    """Customer background information for SPIN generation"""
    industry: str = Field(..., description="客户行业，如：制造业、医疗、教育、零售")
    scale: str = Field(..., description="客户规模，如：中小企业、中大型企业、集团企业")
    pain_points: List[str] = Field(..., description="客户痛点列表")


class SpinQuestionsRequest(BaseModel):
    """Request for SPIN question generation"""
    customer: CustomerContext


class SpinQuestionList(BaseModel):
    """SPIN question list response"""
    question_list_id: str = Field(..., description="问题清单ID")
    situation_questions: List[str] = Field(default_factory=list, description="Situation问题-了解现状")
    problem_questions: List[str] = Field(default_factory=list, description="Problem问题-发现痛点")
    implication_questions: List[str] = Field(default_factory=list, description="Implication问题-放大影响")
    need_payoff_questions: List[str] = Field(default_factory=list, description="Need-payoff问题-引导价值")
    customer_context: CustomerContext
    created_at: str


class SpinQuestionResponse(BaseModel):
    """Response for SPIN question generation"""
    success: bool
    data: Optional[SpinQuestionList] = None
    error: Optional[str] = None
